### PR TITLE
Feat/leveragelp hooks

### PR DESCRIPTION
**Pull request by cryptokage1996**
## Description

Closes:
* https://trello.com/c/yLp67yBg/553-add-hooks-for-tier-calculation-in-leveragelp

### What has Changed?

- Added Hooks for leveragelp
- Added updated_leverage in get-positions-for-address query 

---

### CODE REVIEW

File: x/amm/types/expected_keepers.go [hunk 1]
```diff
@@ -57,3 +57,7 @@ type AssetProfileKeeper interface {
 type AccountedPoolKeeper interface {
 	GetAccountedBalance(sdk.Context, uint64, string) math.Int
 }
+
+type TierKeeper interface {
+	
+}
```
Comment (amityadav0) [inline x/amm/types/expected_keepers.go:63]:
```suggestion
```

File: x/leveragelp/keeper/keeper.go [hunk 2]
```diff
@@ -201,3 +201,14 @@ func (k Keeper) GetWhitelistedAddress(ctx sdk.Context, pagination *query.PageReq
 
 	return list, pageRes, err
 }
+
+// SetHooks set the epoch hooks
```
Comment (amityadav0) [inline x/leveragelp/keeper/keeper.go:205]: need to change the comment here, these hooks are triggered by leveragelp actions

**Review by amityadav0 — COMMENTED** [review]
Please add tests for hooks and updated query, Also we need setHooks in `app.go`

**Comment by amityadav0** [comment]
thanks, could you please add ss of localnet testing ? are hooks working as expected with snapshot ?

**Review by cosmic-vagabond — APPROVED** [review]

**Pull request merged**